Python sdk improvements (#101)
configcat/docs: website/docs/sdk-reference/python.md

```python
def configuration_changed_callback(self):
# Configuration changed.
    pass

configcatclient.create_client_with_auto_poll('#YOUR-SDK-KEY#', on_configuration_changed_callback=configuration_changed_callback)
```
Available options:

| Option Parameter                    | Description                                                                                          | Default |
| ----------------------------------- | ---------------------------------------------------------------------------------------------------- | ------- |
| `poll_interval_seconds`             | Polling interval.                                                                                    | 60      |
| `max_init_wait_time_seconds`        | Maximum waiting time between the client initialization and the first config acquisition in secconds. | 5       |
| `on_configuration_changed_callback` | Callback to get notified about changes.                                                              | -       |
| `config_cache_class`                | Custom cache implementation.                                                                         | None    |

> | `connect_timeout`                   | How many seconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). | 10 |

Reviewer: ```suggestion
| `connect_timeout`                   | The number of seconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). | 10 |
```

> | `read_timeout`                      | How many seconds to wait for the server to send data before giving up.                               | 30      |

Reviewer: ```suggestion
| `read_timeout`                      | The number of seconds to wait for the server to respond before giving up.                               | 30      |
```

| `cache_time_to_live_seconds` | Cache TTL.                   | 60      |
| `config_cache_class`         | Custom cache implementation. | None    |
| `connect_timeout`            | How many seconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). | 10 |
| `read_timeout`               | How many seconds to wait for the server to send data before giving up. | 30 |
| `flagOverrides`              | Configures local feature flag & setting overrides. [More about feature flag overrides](#flag-overrides) | None |

### Manual polling
Manual polling gives you full control over when the `config.json` (with the setting values) is downloaded. *ConfigCat SDK* will not update them automatically. Calling `force_refresh()` is your application's responsibility.

```python
configcat_client = configcatclient.create_client_with_manual_poll('#YOUR-SDK-KEY#')
configcat_client.force_refresh()
```

Available options:

| Option Parameter     | Description                  | Default |
| -------------------- | ---------------------------- | ------- |
| `config_cache_class` | Custom cache implementation. | None    |

> | `connect_timeout`    | How many seconds to wait for the server to make the initial connection (i.e. completing the TCP connection handshake). | 10 |

Reviewer: same as above

> | `read_timeout`       | How many seconds to wait for the server to send data before giving up. | 30 |

Reviewer: same as above

> | `flagOverrides`      | Configures local feature flag & setting overrides. [More about feature flag overrides](#flag-overrides) | None |

Reviewer: Try to avoid the word "configure"
